boskos/client/client.go

```diff
@@ -93,6 +94,31 @@ func (c *Client) Acquire(rtype, state, dest string) (*common.Resource, error) {
 	return r, nil
 }
 
+// AcquireWait blocks until Acquire returns the specified resource or the
+// provided context is cancelled or its deadline exceeded.
+func (c *Client) AcquireWait(ctx context.Context, rtype, state, dest string) (*common.Resource, error) {
+	// Try to acquire the resource until available or the context is
+	// cancelled or its deadline exceeded.
+	if ctx == nil {
```

Comment:
Can we not allow nil context ? 
https://golang.org/pkg/context/, user can pass in context.Background() or TODO().

Reply:
Hi @sebastienvas,

I agree, they can, but since there's no reason to cause a runtime panic here if they **do**, why not avoid the situation? If the context was going to be passed into the actual HTTP request (I have a branch where I played with that), then I'd agree. However, that raises its own concerns (a context deadline shared by both the dial connection attempt and lock acquisition). As it is, sending a nil context results in blocking call with no timeout. I can see how that's advantageous.

Still, I'm not married to being nice about it. I'm also fine with removing the check.

Reply:
Hi @sebastienvas,

I compromised by returning a new error, `ErrContextRequired`, if the AcquireWait variants are invoked with a nil context.